Add the option to view the current branch on GitHub

tsvetilian-ty opened the pull request:
Closes #14224

## Description
Add option to "Branch" to open the current branch on GitHub.

Points to get feedback on before converting draft to standard PR
- [x] Position in the Branch menu
- [x] Should the option have shortcut/accelerator

tsvetilian-ty committed:
Initial implementation of "View branch on GitHub"

tidy-dev commented:
> Position in the Branch menu

Where you put it makes good sense.

> Should the option have shortcut/accelerator

Definitely think it should... however, I am not sure there is a shortcut that just "makes sense". Maybe CtrlOrCmd+Alt+B

Other notes: Would you be willing to update the "Show Pull Request" to "View Pull Request on GitHub" to be consistent with everywhere else in the app as part of this?

tsvetilian-ty committed:
Add accelerator to "View Branch on GitHub""

tsvetilian-ty committed:
Rename "Show Pull Request" to "View Pull Request on GitHub"

tsvetilian-ty commented:
@tidy-dev, thanks for your feedback 💯
I will mark the PR as ready for review

tidy-dev reviewed (dismissed):
Thanks you for contributions @tsvetilian-ty!

(Approving - but also going to give some time before merging so other team members can weigh in on it. 🙂 )

sergiou87 reviewed (dismissed):
I like the consistency in wording and also the new menu item to view the current branch on GH 😍  Thank you!

ericboucher commented:
Wow that was quick, thanks a ton @tidy-dev and @tsvetilian-ty ❤️

ericboucher commented on app/src/ui/app.tsx:
```diff
@@ -612,6 +614,29 @@ export class App extends React.Component<IAppProps, IAppState> {
   }
 
+  private showBranchOnGitHub() {
+    const htmlURL = this.getCurrentRepositoryGitHubURL()
```
Most of the logic in here could be shared with `compareBranchOnDotcom()`

tidy-dev replied:
Good catch! @tsvetilian-ty think you could update this such that the two actions share a method?

tsvetilian-ty replied:
@ericboucher great catch 🔝 @tidy-dev 👍

tsvetilian-ty committed:
Combined showBranchOnGitHub and compareBranchOnDotcom

Combined `showBranchOnGitHub` and `compareBranchOnDotcom` in `openBranchOnGitub` to reduce duplicated code

ericboucher commented on app/src/ui/app.tsx:
```diff
@@ -387,7 +387,9 @@ export class App extends React.Component<IAppProps, IAppState> {
         return this.viewRepositoryOnGitHub()
       case 'compare-on-github':
-        return this.compareBranchOnDotcom()
+        return this.openBranchOnGitub(true)
```
Sorry for being picky but I would call out "compare=true/false" explicitly for clarity

tsvetilian-ty committed:
Merge branch 'desktop:development' into open-current-branch-on-github

tidy-dev commented on app/src/ui/app.tsx:
```diff
@@ -608,8 +610,15 @@ export class App extends React.Component<IAppProps, IAppState> {
 
-    const compareURL = `${htmlURL}/compare/${branchTip.branch.upstreamWithoutRemote}`
-    this.props.dispatcher.openInBrowser(compareURL)
+    let url = htmlURL
```
We try to avoid using `let` if possible and I think we could simplify this function a bit by passing in a variable of `view: 'tree' | 'compare'` and then const url = `${htmlURL}/${view}/${branchTip.branch.upstreamWithoutRemote}`

tidy-dev reviewed (changes requested):
@tsvetilian-ty Thank you for updating that. I think we can make it a tad simpler and easier to read/clarity on action. See my suggestion. :)

tsvetilian-ty committed:
Fix lint issue

tsvetilian-ty committed:
Update openBranchOnGitub signature

tidy-dev reviewed (approved):
✨  Thank so much for those changes!
